docs/architecture/how/cross-shard.md

```diff
@@ -0,0 +1,301 @@
+# Cross shard transactions - deep dive
+
+In this article, we'll look deeper into how cross-shard transactions are working on the simple example of user 'shard0' transfering money to user 'shard1'.
+
+These users are on separate shards (shard0 is on shard 0 and shard1 is on shard 1).
+
+Imagine, we call the following command line:
+
+```shell
+NEAR_ENV=local near send shard0 shard1 500 
+```
+
+What happens under the hood? How is this transaction changed into receipts and processed by near?
+
+
+## From Explorer perspective
+If you look at simple token transfer in explorer (for example: https://explorer.testnet.near.org/transactions/6cNJpNKWP55YrxrgRzc2gi6BM91fo3V6n2vVAo41MvZv), you can see that it is broken into three separate sections:
+
+* convert transaction into receipt ( executed in block B )
+* receipt that transfers tokens ( executed in block B+1 )
+* receipt that refunds gas ( executed in block B+2 )
+
+But under the hood, the situation is a little bit more complex, as there is actually one more receipt (that is created after converting the transaction). Let's take a deeper look.
+
+## Internal perspective (Transactions & Receipts)
+
+One important think to remember, is that NEAR is sharded - so in all our designs, we have to assume that each account is on a separate shard (so that the fact that some of them are colocated doesn't give any advantage).
```

Comment:
```suggestion
One important thing to remember, is that NEAR is sharded - so in all our designs, we have to assume that each account is on a separate shard. So that the fact that some of them are colocated doesn't give any advantage.
```